src/factories/RabbitMQChannelFactory.ts

```diff
@@ -0,0 +1,191 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// noinspection JSMethodCanBeStatic
+
+import {inject, injectable} from 'inversify';
+import {TYPES} from '../constants/types';
+import {AmqpConnectionManager, ChannelWrapper} from 'amqp-connection-manager';
+import {ConfirmChannel, ConsumeMessage} from 'amqplib';
+import {
+    CensusClient,
+    Death,
+    FacilityControl,
+    GainExperience,
+    MetagameEvent,
+    PS2Event,
+    Stream,
+    VehicleDestroy,
+} from 'ps2census';
+import {getLogger} from '../logger';
+import ApplicationException from '../exceptions/ApplicationException';
+import {QueueMessageHandlerInterface} from '../interfaces/QueueMessageHandlerInterface';
+import {Options} from 'amqplib/properties';
+import AdminQueueMessage from '../data/AdminAggregator/AdminQueueMessage';
+
+@injectable()
+export default class RabbitMQChannelFactory {
+    private static readonly logger = getLogger('RabbitMQChannelFactory');
+    private channel: ChannelWrapper;
+
+    constructor(
+        @inject(TYPES.rabbitMqConnection) private readonly rabbit: AmqpConnectionManager,
+        private readonly censusClient: CensusClient,
+    ) {}
+
+    // Above this, QueueAuthority creates these streams
+    public create(
+        exchange: string,
+        queueName: string,
+        queueOptions: Options.AssertQueue = {},
+        pattern: string | null = null,
+        handler: QueueMessageHandlerInterface<any> | null = null,
+    ): ChannelWrapper {
+        this.channel = this.rabbit.createChannel({
```

Comment:
This is a factory, that gets potentially reused, meaning it will overwrite `this.channel` each time. Maybe add it to a map or a set. Though I don't think it is necessary, calling close on the `rabbit` will also close all channel connections.

Reply:
Refacotored into a proper factory pattern here: https://github.com/ps2alerts/aggregator/pull/565/commits/63b80ab605ff19c2e29313f626ae65a1e87607af